src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/MovementResourceTest.kt

```diff
@@ -0,0 +1,418 @@
+package uk.gov.justice.hmpps.prison.api.resource.impl
+
+import net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.http.HttpMethod.GET
+import org.springframework.http.HttpMethod.POST
+import org.springframework.http.HttpStatus.OK
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.web.util.UriComponentsBuilder
+import uk.gov.justice.hmpps.prison.executablespecification.steps.AuthTokenHelper.AuthToken
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalDateTime.now
+import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoUnit
+import java.util.Map
+
+class MovementResourceTest : ResourceTest() {
```

Comment:
converted to kotlin and only added authorisation tests - additional tests exist as cucumber tests, these check the actual values returned